### PR TITLE
Expand unit test coverage: client tests + server edge cases

### LINKED ISSUE #80: Unit tests
## Jobs to Be Done

- As a **contributor**, I want unit tests covering core logic (text anchoring, sanitization, ID generation, URI normalization), so that I can refactor with confidence that I haven't broken anything.
- As a **maintainer**, I want automated test coverage checks, so that new PRs don't reduce code quality.

## Technical Proposal

Expand the existing test suite (currently 15 unit tests for `generate-id`, `normalize-uri`, and `sanitize`) to cover the client-side logic. Extract pure functions from `feedback-layer.js` (selector creation, highlight rendering, comment sorting, DOM manipulation helpers) into testable modules. Use Node's built-in test runner (already in use) for server-side tests. For client-side logic that depends on DOM APIs, use a lightweight DOM mock (jsdom) or test the extracted pure functions directly. Target 80%+ line coverage across server and shared modules. The existing `c8` coverage tooling and `--check-coverage` flag are already configured — extend them to cover new test files.

### PR DESCRIPTION
**Pull request by cass-clearly**
## Summary

- **30 new client-side unit tests** using `node:test` + `node:assert/strict` covering `escapeHtml`, `buildPrompt`, `formatComments`, `threadComments`, `truncate`, `timeAgo`, and `setBaseUrl`
- **3 new server edge-case tests**: POST /comments with nonexistent document ID (404), POST /comments with neither uri nor document (400), POST /documents with invalid URI (400)
- **Extracted `escapeHtml`** as a shared pure utility (`feedback-layer/src/escape-html.js`) — eliminates DOM dependency and deduplicates code between `sidebar.js` and `ui.js`
- **Wired up `c8` coverage** for client tests with 80% line threshold on testable pure modules
- **Total test count**: 89 tests (59 server + 30 client), all passing

## What changed and why

The server had 46+ tests with 80% coverage enforced, but the client had zero tests and zero test infrastructure. Several client modules contain pure functions that are trivially testable without a DOM. This PR adds that missing coverage.

The `_testExports` pattern exposes internal pure functions for testing without polluting the public API. Coverage is scoped to fully-testable files (`escape-html.js`, `prompt-builder.js`) to avoid false failures from DOM-dependent code.

## How to verify

```bash
export DATABASE_URL=postgres://remarq:remarq@localhost:5433/remarq
npm test          # runs both server + client suites with coverage
npm run build     # verify bundle still builds
```

Closes #80

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: feedback-layer/src/prompt-builder.js [hunk 1]
```diff
@@ -79,3 +79,5 @@ function formatComments(comments) {
     })
     .join("\n\n");
 }
+
+export const _testExports = { formatComments };
```
Comment (csalvato) [inline feedback-layer/src/prompt-builder.js]: Shouldn't `formatComments` be in it's own file and exported from that file and imported into this one?

File: feedback-layer/package-lock.json
Comment (csalvato) [inline feedback-layer/package-lock.json]: Looks unrelated to testing. Pull into a separate PR?

File: CLAUDE.md [hunk 2]
```diff
@@ -44,3 +44,21 @@ If the review agent requests changes, fix them on the same branch, push, and re-
...
+- **Client tests:** `npm run test:client`
+- **All tests:** `npm test`
+
+Both suites use `node:test` + `node:assert/strict` and `c8` for coverage with 80% line threshold.
+
+### `_testExports` pattern
+Client modules export internal pure functions via a `_testExports` named export for unit testing without a DOM:
+```js
+export const _testExports = { threadComments, truncate, timeAgo };
+```
```
Comment (csalvato) [inline CLAUDE.md]: We shouldn't do this. We should instead split the files into separate utility functions in a seprarate directory that we import both into our files and the test files for testing.
